### PR TITLE
18 create cohorts page

### PR DESCRIPTION
**Pull request by KhaledAref13**

### CODE REVIEW

**Review by Nour-Agha99 — APPROVED** [review]

**Review by mobatra — APPROVED** [review]

File: src/components/Cohorts/CohortCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  startDate: string,
+  endDate: string,
+}
+
+const CohortCard = ({
+  alt, imgSrc, cohortName, cohortMembers, startDate, endDate,
+}: IProps) => (
+  <Grid item xs={15} sm={5} md={5} lg={3}>
+    <CohortCardDiv>
+      <CohortCardImgDiv>
```
Comment (karamzomlot) [inline src/components/Cohorts/CohortCard/index.tsx]: Could you make this component when you click it on redirect you to the cohort page?
hint (useNavigate hook from `react-router-dom`)

File: src/components/Cohorts/CohortCard/style.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+const CohortCardDiv = styled('div')`
+  background-color: #fff;
+  border-radius: 0.3rem;
+  border: 1px solid transparent;
+  box-shadow: 0 0 15px 0 rgba(82, 63, 105, 0.05);
+  transition: 0.3s;
+  border-bottom: 3px solid transparent;
+  &:hover {
+    transform: translateY(-5px);
+    border-bottom-color: #0abb87;
```
Comment (karamzomlot) [inline src/components/Cohorts/CohortCard/style.ts]: I think the hover will be better without the bottom border Coz we have a button (visit)

File: src/components/Cohorts/CohortCard/style.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  font-size: 0.9rem;
+  font-weight: 500;
+`;
+
+const VisitBtn = styled(Button)`
+  background-color: rgba(0, 0, 20, 0.05);
+  color: #6c7293;
+  margin-top: 20px;
+  width: 100%;
+  padding: 4px;
```
Comment (karamzomlot) [inline src/components/Cohorts/CohortCard/style.ts]: Increasing the padding will be better e.g.: (`padding: .6rem 1rem;`)

File: src/components/Cohorts/CohortCard/style.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+  &:hover {
+    transform: translateY(-5px);
+    border-bottom-color: #0abb87;
+  }
+`;
+
+const CohortCardImgDiv = styled('div')`
+  width: 100%;
+  height: 150px;
+  background: url('../../../public/cohort-background.jpg') center center;
```
Comment (karamzomlot) [inline src/components/Cohorts/CohortCard/style.ts]: When you have any file in the public folder you can access it with an absolute  path like this (`/public/file.jpg`).

One more thing if you can change the image to any image from `GSG` will be more beautiful. I will leave to you edited style for this component with adding glass effect

```tsx
const CohortCardImgDiv = styled('div')`
  width: 100%;
  height: 150px;
  background: url('https://mlwhex1ecbob.i.optimole.com/w:1814/h:1102/q:mauto/f:avif/https://gazaskygeeks.com/wp-content/uploads/2020/08/Copy-of-GAT1038.jpg') center center;
  background-repeat: no-repeat;
  background-size: cover;
  border-top-right-radius: 0.3rem;
  border-top-left-radius: 0.3rem;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  position: relative;

  &::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background-color: #4a4a4a8c;
    backdrop-filter: blur(5px);
    border-top-right-radius: 0.3rem;
    border-top-left-radius: 0.3rem;
  }
`;
```
But Coz I used position absolute and relative you should edit the avatar style with adding `z-index: 1`.

```tsx
<Avatar
  alt={alt}
  src={imgSrc}
  sx={{
  width: 90, height: 90, boxShadow: '0 0 0 5px #ffffff', zIndex: 1,
  }}
/>
```

File: src/components/Cohorts/CohortsTopBar/LoadingButton.tsx [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+          padding: '0.5rem 1rem',
+          fontFamily: 'Poppins',
+          textTransform: 'capitalize',
+          fontWeight: '600',
+          opacity: '1',
+          borderRadius: '0.3rem',
+          position: 'relative',
+        }}
+        // eslint-disable-next-line react/jsx-no-bind
+        onClick={handleClick}
```
Comment (karamzomlot) [inline src/components/Cohorts/CohortsTopBar/LoadingButton.tsx]: It's not best practice to disable eslint in your case you can use this 
```tsx
onClick={() => handleClick()}
```

File: src/components/Cohorts/CohortsTopBar/LoadingButton.tsx [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+        startIcon={<FaUserFriends />}
```
Comment (karamzomlot) [inline src/components/Cohorts/CohortsTopBar/LoadingButton.tsx]: To make the component reusable you can add the `startIcon` as a prop and use the button on all pages by adding the text and the icon, other props

File: src/pages/user/Cohorts.tsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import {
+  CallToAction,
+  CohortTopBar,
+  CohortsWrapper,
+  PageTitle,
+  Pagination,
+} from '../../components';
+
+const CohortsContainer = styled.div`
+  padding: calc(70px + 1rem) 1rem 1rem;
```
Comment (karamzomlot) [inline src/pages/user/Cohorts.tsx]: Keep this padding in the outlet,

`layout.tsx`
```tsx
const OutletContainer = styled.div`
  width: 100%;
  padding-top: calc(70px);
`;
```

`Cohorts.tsx`
```tsx
const CohortsContainer = styled.div`
  padding: 1rem 1rem;
`;
```

File: src/pages/user/indext.ts [hunk 8]
```diff
@@ -0,0 +1,3 @@
+import Cohorts from './Cohorts';
+
+export default Cohorts;
```
Comment (karamzomlot) [inline src/pages/user/indext.ts]: Why do you have 2 index files in the same folder
hint this file is `indext.ts` not `index.ts`

and you have an `index.ts` file so you can delete this file and use another one

File: src/router/index.tsx [hunk 9]
```diff
@@ -41,7 +41,7 @@ const router = createBrowserRouter([
         element: <div>Home</div>,
       },
       { path: 'community', element: <div>Community</div> },
-      { path: 'cohorts', element: <div>Cohorts</div> },
+      { path: 'cohorts', element: <Cohorts /> },
```
Comment (karamzomlot) [inline src/router/index.tsx:44]: Add single cohort page route
```tsx
{ path: 'cohorts', element: <Cohorts /> },
{ path: 'cohorts/:cohortName', element: <div>Single Cohort</div> },
```

**Review by karamzomlot — CHANGES_REQUESTED** [review]
Delete any empty `style.ts` files  (`/src/components/Cohorts/CohortsWrapper/style.ts`, `/src/components/Cohorts/CohortMembersTopBar/style.ts`)

File: src/components/Cohorts/CohortsTopBar/LoadingButton.tsx [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+  setTimeout(() => {
+    setLoading(false);
+  }, 3000);
+  return (
+    <Box sx={{ '& > button': { m: 0 } }}>
+      <LoadingButton
+        sx={{
+          backgroundColor: '#4E64DD',
+          'border-color': 'rgba(0,0,0,0)',
```
Comment (karamzomlot) [inline src/components/Cohorts/CohortsTopBar/LoadingButton.tsx]: borderColor not 'border-color` 
If you open the browser console you will notice this error

**Review by Shatha-Amin — APPROVED** [review]

**Review by mobatra — APPROVED** [review]

File: src/components/Cohorts/CohortCard/index.tsx (hunk 1 above)
Comment (karamzomlot) [reply src/components/Cohorts/CohortCard/index.tsx]: @KhaledAref13 
Here I mean `CohortCardImgDiv` when clicking on redirects you to the single cohort

File: src/components/Cohorts/CohortsTopBar/LoadingButton.tsx (hunk 5 above)
Comment (karamzomlot) [reply src/components/Cohorts/CohortsTopBar/LoadingButton.tsx]: @KhaledAref13 
You edit this but forget to delete the comment (disable eslint)

File: src/components/Cohorts/CohortsTopBar/LoadingButton.tsx (hunk 10 above)
Comment (karamzomlot) [reply src/components/Cohorts/CohortsTopBar/LoadingButton.tsx]: @KhaledAref13 
Nothing is updated here.

**Review by karamzomlot — APPROVED** [review]

**Pull request merged**